src/Compilers/CSharp/Test/Semantic/Semantics/InterpolationTests.cs

```diff
@@ -13865,30 +13913,78 @@ static CustomHandler F1()
     {
         Span<char> s = stackalloc char[10];
         CustomHandler h1 = $""{s}"";
-        return h1;
+        return h1; // 1
     }
     static CustomHandler F2()
     {
         Span<char> s = stackalloc char[10];
         CustomHandler h2 = new CustomHandler(0, 1);
-        h2.AppendFormatted(s);
+        h2.AppendFormatted(s); // 2
         return h2;
     }
 }
 ";
             var comp = CreateCompilation(new[] { code, InterpolatedStringHandlerAttribute }, parseOptions: TestOptions.Regular.WithLanguageVersion(languageVersion), targetFramework: TargetFramework.NetCoreApp);
             comp.VerifyDiagnostics(
                 // (15,16): error CS8352: Cannot use variable 'h1' in this context because it may expose referenced variables outside of their declaration scope
-                //         return h1;
+                //         return h1; // 1
                 Diagnostic(ErrorCode.ERR_EscapeVariable, "h1").WithArguments("h1").WithLocation(15, 16),
                 // (21,9): error CS8350: This combination of arguments to 'CustomHandler.AppendFormatted(Span<char>)' is disallowed because it may expose variables referenced by parameter 's' outside of their declaration scope
-                //         h2.AppendFormatted(s);
+                //         h2.AppendFormatted(s); // 2
                 Diagnostic(ErrorCode.ERR_CallArgMixing, "h2.AppendFormatted(s)").WithArguments("CustomHandler.AppendFormatted(System.Span<char>)", "s").WithLocation(21, 9),
                 // (21,28): error CS8352: Cannot use variable 's' in this context because it may expose referenced variables outside of their declaration scope
-                //         h2.AppendFormatted(s);
+                //         h2.AppendFormatted(s); // 2
                 Diagnostic(ErrorCode.ERR_EscapeVariable, "s").WithArguments("s").WithLocation(21, 28));
         }
 
+        // As above but with scoped parameter in AppendFormatted().
+        [WorkItem(63262, "https://github.com/dotnet/roslyn/issues/63262")]
+        [Fact]
+        public void RefEscape_12B()
+        {
+            var code =
+@"using System;
+using System.Runtime.CompilerServices;
+[InterpolatedStringHandler]
+ref struct CustomHandler
+{
+    public CustomHandler(int literalLength, int formattedCount) { }
+    public void AppendFormatted(scoped Span<char> s) { }
+}
+class Program
+{
+    static CustomHandler F1()
+    {
+        Span<char> s = stackalloc char[10];
+        CustomHandler h1 = $""{s}"";
+        return h1; // 1
```

Comment:
Why is this occurring? The `AppendFormatted` call is marked `scoped`, so the `safe-to-escape` of `h1` should not be restricted to this method. This _will_ break the BCL and other libraries if it is merged as is.

Reply:
Good catch, thanks.